rule/rule.go

```diff
@@ -816,6 +816,27 @@ func (r *Rule) sync() {
 	r.updated = false
 }
 
+// AttrEqual decides if the rule and the other rule has the same value on the attribute
+func (r *Rule) AttrEqual(other *Rule, key string) bool {
```

Comment:
Let's move this into `merger` as `func attrsMatch(x, y *Rule, key string) bool`. Everything in `rule` needs to be pretty general. This is pretty targeted toward matching and merging, so I think it should be in `merger`.

I'm not sure if it makes sense to require that the two lists are equal. If a source file is added or deleted, the rules will stop matching. Perhaps it would be better to check whether at least one element matches? If you want to leave it as-is for now though, we can always loosen it later.

Reply:
To handle added and delete source files, we can check if one list is a subset of the other, but I am not sure how to handle renames